In [44]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [59]:
df = pd.read_csv('data/train_to_download_final_1.csv')
df.head()

galactic year                        galaxy  existence expectancy index  \
0         990025  Large Magellanic Cloud (LMC)                    0.628657   
1         990025              Camelopardalis B                    0.818082   
2         990025                       Virgo I                    0.659443   
3         990025            UGC 8651 (DDO 181)                    0.555862   
4         990025                  Tucana Dwarf                    0.991196   

   existence expectancy at birth  Gross income per capita  Income Index  \
0                      63.125200             27109.234310      0.646039   
1                      81.004994             30166.793958      0.852246   
2                      59.570534              8441.707353      0.499762   
3                      52.333293                      NaN           NaN   
4                      81.802464             81033.956906      1.131163   

   Expected years of education (galactic years)  \
0                                      8.240543   
1                                     10.671823   
2                                      8.840316   
3                                           NaN   
4                                     13.800672   

   Mean years of education (galactic years)  \
0                                       NaN   
1                                  4.742470   
2                                  5.583973   
3                                       NaN   
4                                 13.188907   

   Intergalactic Development Index (IDI)  Education Index  ...  \
0                                    NaN              NaN  ...   
1                               0.833624         0.467873  ...   
2                               0.469110         0.363837  ...   
3                                    NaN              NaN  ...   
4                               0.910341         0.918353  ...   

   Intergalactic Development Index (IDI), female  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4                                            NaN   

   Intergalactic Development Index (IDI), male  \
0                                          NaN   
1                                          NaN   
2                                          NaN   
3                                          NaN   
4                                          NaN   

   Gender Development Index (GDI)  \
0                             NaN   
1                             NaN   
2                             NaN   
3                             NaN   
4                             NaN   

   Intergalactic Development Index (IDI), female, Rank  \
0                                                NaN     
1                                                NaN     
2                                                NaN     
3                                                NaN     
4                                                NaN     

   Intergalactic Development Index (IDI), male, Rank  Adjusted net savings   \
0                                                NaN                    NaN   
1                                                NaN              19.177926   
2                                                NaN              21.151265   
3                                                NaN                    NaN   
4                                                NaN                    NaN   

   Creature Immunodeficiency Disease prevalence, adult (% ages 15-49), total  \
0                                                NaN                           
1                                                NaN                           
2                                           6.534020                           
3                                           5.912194                           
4                                  

# Предобработка
По мере реализации проекта перенесем предобработку в отдельные файлы

### Названия столбцов
Названия у столбцов конечно интересные, однако, в реальности они очень мешают. Так-что будем, по мере понимая содержания сталюцов, заменять их на более адекватные и короткие.

In [60]:
df = df.rename(columns={"galactic year": "year", "galaxy": "country"})
df.head(2)

year                       country  existence expectancy index  \
0  990025  Large Magellanic Cloud (LMC)                    0.628657   
1  990025              Camelopardalis B                    0.818082   

   existence expectancy at birth  Gross income per capita  Income Index  \
0                      63.125200             27109.234310      0.646039   
1                      81.004994             30166.793958      0.852246   

   Expected years of education (galactic years)  \
0                                      8.240543   
1                                     10.671823   

   Mean years of education (galactic years)  \
0                                       NaN   
1                                   4.74247   

   Intergalactic Development Index (IDI)  Education Index  ...  \
0                                    NaN              NaN  ...   
1                               0.833624         0.467873  ...   

   Intergalactic Development Index (IDI), female  \
0                                            NaN   
1                                            NaN   

   Intergalactic Development Index (IDI), male  \
0                                          NaN   
1                                          NaN   

   Gender Development Index (GDI)  \
0                             NaN   
1                             NaN   

   Intergalactic Development Index (IDI), female, Rank  \
0                                                NaN     
1                                                NaN     

   Intergalactic Development Index (IDI), male, Rank  Adjusted net savings   \
0                                                NaN                    NaN   
1                                                NaN              19.177926   

   Creature Immunodeficiency Disease prevalence, adult (% ages 15-49), total  \
0                                                NaN                           
1                                                NaN                           

   Private galaxy capital flows (% of GGP)  Gender Inequality Index (GII)  \
0                                      NaN                            NaN   
1                                22.785018                            NaN   

          y  
0  0.052590  
1  0.059868  

[2 rows x 80 columns]

### Формат даты

In [61]:
df['year'].unique()

array([ 990025,  991020,  992016,  993012,  994009,  995006,  996004,
        997002,  998001,  999000, 1000000, 1001000, 1002001, 1003002,
       1004004, 1005006, 1006009, 1007012, 1008016, 1009020, 1010025,
       1011030, 1012036, 1013042, 1014049, 1015056], dtype=int64)

Если присмотреться к датам и немного подумать, то приходишь к выводу о том, что первые 3 цифры + 1000 = год н.э. Например 990025 => 1000 + 990 => 1990. А так-как мы имеем дело с макроэкономическими показателями, то последние 2 цифры можно отбросить.

In [62]:
df['year'] = df['year']//1000+1000
df.head()

year                       country  existence expectancy index  \
0  1990  Large Magellanic Cloud (LMC)                    0.628657   
1  1990              Camelopardalis B                    0.818082   
2  1990                       Virgo I                    0.659443   
3  1990            UGC 8651 (DDO 181)                    0.555862   
4  1990                  Tucana Dwarf                    0.991196   

   existence expectancy at birth  Gross income per capita  Income Index  \
0                      63.125200             27109.234310      0.646039   
1                      81.004994             30166.793958      0.852246   
2                      59.570534              8441.707353      0.499762   
3                      52.333293                      NaN           NaN   
4                      81.802464             81033.956906      1.131163   

   Expected years of education (galactic years)  \
0                                      8.240543   
1                                     10.671823   
2                                      8.840316   
3                                           NaN   
4                                     13.800672   

   Mean years of education (galactic years)  \
0                                       NaN   
1                                  4.742470   
2                                  5.583973   
3                                       NaN   
4                                 13.188907   

   Intergalactic Development Index (IDI)  Education Index  ...  \
0                                    NaN              NaN  ...   
1                               0.833624         0.467873  ...   
2                               0.469110         0.363837  ...   
3                                    NaN              NaN  ...   
4                               0.910341         0.918353  ...   

   Intergalactic Development Index (IDI), female  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4                                            NaN   

   Intergalactic Development Index (IDI), male  \
0                                          NaN   
1                                          NaN   
2                                          NaN   
3                                          NaN   
4                                          NaN   

   Gender Development Index (GDI)  \
0                             NaN   
1                             NaN   
2                             NaN   
3                             NaN   
4                             NaN   

   Intergalactic Development Index (IDI), female, Rank  \
0                                                NaN     
1                                                NaN     
2                                                NaN     
3                                                NaN     
4                                                NaN     

   Intergalactic Development Index (IDI), male, Rank  Adjusted net savings   \
0                                                NaN                    NaN   
1                                                NaN              19.177926   
2                                                NaN              21.151265   
3                                                NaN                    NaN   
4                                                NaN                    NaN   

   Creature Immunodeficiency Disease prevalence, adult (% ages 15-49), total  \
0                                                NaN                           
1                                                NaN                           
2                                           6.534020                           
3                                           5.912194                           
4                                           5.611753                           

   Priva

# Черновик

In [63]:
df_test = pd.read_csv('data/test_to_download_final_1.csv')
df_test = df_test.rename(columns={"galactic year": "year", "galaxy": "country"})
df_test['year'] = df_test['year']//1000+1000
df_test.head()

year        country  existence expectancy index  \
0  2007        KK98 77                    0.456086   
1  2007  Reticulum III                    0.529835   
2  2008  Reticulum III                    0.560976   
3  2007        Segue 1                    0.565910   
4  2013        Virgo I                    0.588274   

   existence expectancy at birth  Gross income per capita  Income Index  \
0                      51.562543             12236.576447      0.593325   
1                      57.228262              3431.883825      0.675407   
2                      59.379539             27562.914252      0.594624   
3                      59.952390             20352.232905      0.837700   
4                      55.428320             23959.704016      0.520579   

   Expected years of education (galactic years)  \
0                                     10.414164   
1                                      7.239485   
2                                     11.774890   
3                                     11.613621   
4                                     10.392416   

   Mean years of education (galactic years)  \
0                                 10.699072   
1                                  5.311122   
2                                  5.937797   
3                                 10.067882   
4                                  6.374637   

   Intergalactic Development Index (IDI)  Education Index  ...  \
0                               0.547114         0.556267  ...   
1                               0.497688         0.409969  ...   
2                               0.544744         0.486167  ...   
3                               0.691641         0.523441  ...   
4                               0.530676         0.580418  ...   

   Current health expenditure (% of GGP)  \
0                                    NaN   
1                                    NaN   
2                                    NaN   
3                                    NaN   
4                               7.357729   

   Intergalactic Development Index (IDI), female  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4                                       0.583373   

   Intergalactic Development Index (IDI), male  \
0                                          NaN   
1                                          NaN   
2                                          NaN   
3                                          NaN   
4                                     0.600445   

   Gender Development Index (GDI)  \
0                             NaN   
1                             NaN   
2                             NaN   
3                             NaN   
4                        0.856158   

   Intergalactic Development Index (IDI), female, Rank  \
0                                                NaN     
1                                                NaN     
2                                                NaN     
3                                                NaN     
4                                         206.674424     

   Intergalactic Development Index (IDI), male, Rank  Adjusted net savings   \
0                                                NaN                    NaN   
1                                                NaN                    NaN   
2                                                NaN                    NaN   
3                                                NaN                    NaN   
4                                         224.104054                    NaN   

   Creature Immunodeficiency Disease prevalence, adult (% ages 15-49), total  \
0                                                NaN                           
1                                                NaN                           
2                                                NaN                      

In [67]:
df[df['country']=='Large Magellanic Cloud (LMC)']

year                       country  existence expectancy index  \
0     1990  Large Magellanic Cloud (LMC)                    0.628657   
194   1991  Large Magellanic Cloud (LMC)                    0.627245   
410   1992  Large Magellanic Cloud (LMC)                    0.662135   
543   1993  Large Magellanic Cloud (LMC)                    0.687776   
765   1994  Large Magellanic Cloud (LMC)                    0.727717   
1053  1995  Large Magellanic Cloud (LMC)                    0.801878   
1244  1996  Large Magellanic Cloud (LMC)                    0.727336   
1411  1997  Large Magellanic Cloud (LMC)                    0.751123   
1570  1998  Large Magellanic Cloud (LMC)                    0.817002   
1758  1999  Large Magellanic Cloud (LMC)                    0.765157   
1833  2000  Large Magellanic Cloud (LMC)                    0.820034   
2015  2001  Large Magellanic Cloud (LMC)                    0.702311   
2274  2002  Large Magellanic Cloud (LMC)                    0.728519   
2421  2003  Large Magellanic Cloud (LMC)                    0.776939   
2611  2004  Large Magellanic Cloud (LMC)                    0.756422   
2846  2005  Large Magellanic Cloud (LMC)                    0.883358   
2891  2006  Large Magellanic Cloud (LMC)                    0.881534   
3679  2013  Large Magellanic Cloud (LMC)                    0.814879   
3770  2014  Large Magellanic Cloud (LMC)                    0.874476   

      existence expectancy at birth  Gross income per capita  Income Index  \
0                         63.125200             27109.234310      0.646039   
194                       62.389059             17114.109182      0.503198   
410                       68.794245             13225.033915      0.692653   
543                       62.848660              9116.133706      0.667264   
765                       67.482431              3598.172394      0.609830   
1053                      64.024931             30050.765207      0.560245   
1244                      61.110448             16921.403977      0.731106   
1411                      71.737634             14009.050465      0.695900   
1570                      67.396675             12586.331228      0.675266   
1758                      68.828983             18638.673092      0.605786   
1833                      71.804603             21048.858683      0.612216   
2015                      68.111525             14334.420826      0.722256   
2274                      69.722603             20028.321235      0.787673   
2421                      75.038383             19026.152154      0.673346   
2611                      76.167342             22050.589590      0.909053   
2846                      81.251875             21785.494822      0.767434   
2891                      70.398402             17220.027191      0.823692   
3679                      83.287364             16453.884262      0.899114   
3770                      73.896443             24825.701169      0.848819   

      Expected years of education (galactic years)  \
0                                         8.240543   
194                                       9.317262   
410                                       8.030645   
543                                      11.347429   
765                                      10.402265   
1053                                      8.811260   
1244                                      9.915121   
1411                                      9.231937   
1570                                      8.640981   
1758                                      9.540513   
1833                                     11.263948   
2015                                      9.684950   
2274                                     12.043517   
2421                                     12.649409   
2611                                     13.051400   
2846                                     14.378038   
2891                                     12.883945   
3679                            

In [68]:
df_test[df_test['country']=='Large Magellanic Cloud (LMC)']

year                       country  existence expectancy index  \
144  2008  Large Magellanic Cloud (LMC)                    0.779758   
160  2015  Large Magellanic Cloud (LMC)                    0.790034   
238  2009  Large Magellanic Cloud (LMC)                    0.844595   
256  2012  Large Magellanic Cloud (LMC)                    0.853172   
309  2007  Large Magellanic Cloud (LMC)                    0.884930   
322  2016  Large Magellanic Cloud (LMC)                    0.890742   
375  2010  Large Magellanic Cloud (LMC)                    0.918996   
422  2011  Large Magellanic Cloud (LMC)                    0.937656   

     existence expectancy at birth  Gross income per capita  Income Index  \
144                      75.346249             19231.348529      0.707828   
160                      79.425296             21972.122930      0.831417   
238                      73.130032             24127.802732      0.833485   
256                      77.321986             27416.106737      0.880479   
309                      76.399821             21735.014557      0.877009   
322                      84.311153             25033.664800      0.702787   
375                      78.142755             28751.348361      0.853777   
422                      83.695703             21567.525973      0.876733   

     Expected years of education (galactic years)  \
144                                     13.327868   
160                                     17.095940   
238                                     14.733714   
256                                     18.084904   
309                                     13.377857   
322                                     16.539201   
375                                     14.649964   
422                                     14.426615   

     Mean years of education (galactic years)  \
144                                  6.806029   
160                                  6.481246   
238                                  6.837942   
256                                  6.901763   
309                                  8.086796   
322                                  4.821879   
375                                  5.930653   
422                                  4.763673   

     Intergalactic Development Index (IDI)  Education Index  ...  \
144                               0.672352         0.586943  ...   
160                               0.686543         0.597843  ...   
238                               0.748864         0.643202  ...   
256                               0.826286         0.654600  ...   
309                               0.738597         0.473309  ...   
322                               0.820799         0.597755  ...   
375                               0.721419         0.494747  ...   
422                               0.809155         0.577366  ...   

     Current health expenditure (% of GGP)  \
144                                    NaN   
160                               7.608230   
238                                    NaN   
256                               5.974166   
309                                    NaN   
322                               5.230947   
375                               4.511014   
422                               5.952772   

     Intergalactic Development Index (IDI), female  \
144                                            NaN   
160                                       0.726385   
238                                            NaN   
256                                       0.593091   
309                                            NaN   
322                                       0.732648   
375                                       0.736388   
422                                       0.639062   

     Intergalactic Development Index (IDI), male  \
144                                          NaN   
160                                     0.776223   
238                                          NaN   
256                         

In [51]:
df[df['country']=='Large Magellanic Cloud (LMC)']

year                       country  existence expectancy index  \
0     1990  Large Magellanic Cloud (LMC)                    0.628657   
194   1991  Large Magellanic Cloud (LMC)                    0.627245   
410   1992  Large Magellanic Cloud (LMC)                    0.662135   
543   1993  Large Magellanic Cloud (LMC)                    0.687776   
765   1994  Large Magellanic Cloud (LMC)                    0.727717   
1053  1995  Large Magellanic Cloud (LMC)                    0.801878   
1244  1996  Large Magellanic Cloud (LMC)                    0.727336   
1411  1997  Large Magellanic Cloud (LMC)                    0.751123   
1570  1998  Large Magellanic Cloud (LMC)                    0.817002   
1758  1999  Large Magellanic Cloud (LMC)                    0.765157   
1833  2000  Large Magellanic Cloud (LMC)                    0.820034   
2015  2001  Large Magellanic Cloud (LMC)                    0.702311   
2274  2002  Large Magellanic Cloud (LMC)                    0.728519   
2421  2003  Large Magellanic Cloud (LMC)                    0.776939   
2611  2004  Large Magellanic Cloud (LMC)                    0.756422   
2846  2005  Large Magellanic Cloud (LMC)                    0.883358   
2891  2006  Large Magellanic Cloud (LMC)                    0.881534   
3679  2013  Large Magellanic Cloud (LMC)                    0.814879   
3770  2014  Large Magellanic Cloud (LMC)                    0.874476   

      existence expectancy at birth  Gross income per capita  Income Index  \
0                         63.125200             27109.234310      0.646039   
194                       62.389059             17114.109182      0.503198   
410                       68.794245             13225.033915      0.692653   
543                       62.848660              9116.133706      0.667264   
765                       67.482431              3598.172394      0.609830   
1053                      64.024931             30050.765207      0.560245   
1244                      61.110448             16921.403977      0.731106   
1411                      71.737634             14009.050465      0.695900   
1570                      67.396675             12586.331228      0.675266   
1758                      68.828983             18638.673092      0.605786   
1833                      71.804603             21048.858683      0.612216   
2015                      68.111525             14334.420826      0.722256   
2274                      69.722603             20028.321235      0.787673   
2421                      75.038383             19026.152154      0.673346   
2611                      76.167342             22050.589590      0.909053   
2846                      81.251875             21785.494822      0.767434   
2891                      70.398402             17220.027191      0.823692   
3679                      83.287364             16453.884262      0.899114   
3770                      73.896443             24825.701169      0.848819   

      Expected years of education (galactic years)  \
0                                         8.240543   
194                                       9.317262   
410                                       8.030645   
543                                      11.347429   
765                                      10.402265   
1053                                      8.811260   
1244                                      9.915121   
1411                                      9.231937   
1570                                      8.640981   
1758                                      9.540513   
1833                                     11.263948   
2015                                      9.684950   
2274                                     12.043517   
2421                                     12.649409   
2611                                     13.051400   
2846                                     14.378038   
2891                                     12.883945   
3679                            